### PR TITLE
upgrade to hyperkube-arm:v1.2.0, docker >1.8 comp

### PR DESCRIPTION
**Pull request by madmas**
changes to get the scripts working with the hypriot-rpi-20160306-192317.img RaspberryPi Images and recent hypercube-arm version.

upgrade hypercube-arm version used in scripts
replace docker's deprecated '-d' with 'daemon'

thanks to @sclassen and @moschkom for the great code camp

### CODE REVIEW

File: rootfs/lib/systemd/system/docker-bootstrap.service [hunk 1]
```diff
@@ -5,7 +5,7 @@ After=network.target docker-bootstrap.socket
 Requires=docker-bootstrap.socket
 
 [Service]
-ExecStart=/usr/bin/docker daemon -H unix:///var/run/docker-bootstrap.sock -p /var/run/docker-bootstrap.pid --storage-driver=overlay --storage-opt dm.basesize=10G --iptables=false --ip-masq=false --bridge=none --graph=/var/lib/docker-bootstrap
+ExecStart=/usr/bin/docker daemon -H unix:///var/run/docker-bootstrap.sock -p /var/run/docker-bootstrap.pid --storage-driver=overlay --storage-opt dm.basesize=10G --iptables=false --ip-masq=false --bridge=none --graph=/var/lib/docker-bootstrap
```
Comment (nicolaiskogheim) [inline rootfs/lib/systemd/system/docker-bootstrap.service]: Is this change only introducing a trailing space? :)

**Comment by madmas** [comment]
@nicolaiskogheim yes, there was only the trailing space in the line, thanks for pointing it out, I removed it

**Comment by nicolaiskogheim** [comment]
I have been using this branch for a while now and it works great on `Hypriot Version 0.7.0 Berry (beta)`. The docker version there is `1.10.2, build c3959b1`

However, on `Hypriot Version 0.8.0 Barbossa`(Docker version `1.11.1, build 5604cbe`), the `k8s-worker.service` won't start, and I'm tearing my hair out trying to figure out why.

~~`systemd` fails because docker tries to stop and delete non-running, non-existing containers, even though [`ExecStartPre`](https://github.com/awassink/k8s-on-rpi/blob/master/rootfs/lib/systemd/system/k8s-worker.service#L7-L8) (where it happens) is defined as to not fail even if the command returns a non-zero exit status. The syntax for that is `ExecStartPre=-command`, where the `-`-sign tells `systemd` to not care about exit status.~~
(It did actually ignore that error.)

`systemd` may not be at fault here, as just trying to run Hyperkube fails.
For instance `docker run -it gcr.io/google_containers/hyperkube-arm:v1.2.0 /hyperkube` fails with a message saying that the command `/hyperkube` doesn't exist within the container. This error does not occur on Hypriot 0.7.0 Berry.

I guess this warrants a bug report somewhere. Just wanted to share my thoughts.

Just as I was finished writing this, @noh4ck suggested replacing the docker 1.11.1 binary with a 1.10.2 one, and suddenly I was able to use Hyperkube again.
So the conclusion is that we probably can't do anything about it in this repo, but know that this installation won't work on Hypriot 0.8.0 Barbossa.

Edit: related: https://github.com/kubernetes/kubernetes/issues/26038#issuecomment-232765600

> I had the same problem running k8s on Hypriot. I had to add cgroup_enable=cpuset to /boot/cmdline.txt

**Comment by Schweigi** [comment]
@nicolaiskogheim I got another issue with Docker v1.11 too besides the `cpuset` problem. I fixed it in my PR https://github.com/awassink/k8s-on-rpi/pull/7 by appending `--exec-root=/var/lib/docker-bootstrap` to `docker-boostrap.service`.

**Comment by nicolaiskogheim** [comment]
@Schweigi Ah, sweet. I regularly have hick-ups with docker on rpi. I will try out your PR soon!

**Pull request closed without merging**